File: typescript/tests/integration/beta-skills-get-metadata.integration.test.ts
/**
 * Integration tests for BetaSkills.getMetadata() via GetSkillMetaData POP action.
 *
 * Note: This test calls the real backend. Do not run concurrently.
 */

import { AgentBay } from "../../src/agent-bay";
import { log } from "../../src/utils/logger";

describe("Beta skills getMetadata integration tests", () => {
  const apiKey = process.env.AGENTBAY_API_KEY;
  let agentBay: AgentBay;

  beforeEach(() => {
    if (apiKey) {
      agentBay = new AgentBay({ apiKey });
    }
  });

  test("should return skillsRootPath and skills list", async () => {
    if (!apiKey) {
      log("Skipping test: AGENTBAY_API_KEY not set");
      return;
    }

    const result = await agentBay.betaSkills.getMetadata();

    expect(result).toBeDefined();
    expect(typeof result.skillsRootPath).toBe("string");
    expect(result.skillsRootPath.length).toBeGreaterThan(0);
    expect(Array.isArray(result.skills)).toBe(true);
  }, 30000);

  test("should filter by group IDs", async () => {
    if (!apiKey) {
      log("Skipping test: AGENTBAY_API_KEY not set");
      return;
    }

    const result = await agentBay.betaSkills.getMetadata({
      skillNames: ["5kvAvffm"],
    });

    expect(result).toBeDefined();
    expect(Array.isArray(result.skills)).toBe(true);
    expect(typeof result.skillsRootPath).toBe("string");
  }, 30000);

  test("should accept imageId parameter", async () => {
    if (!apiKey) {
      log("Skipping test: AGENTBAY_API_KEY not set");
      return;
    }

    const result = await agentBay.betaSkills.getMetadata({
      imageId: "linux_latest",
    });

    expect(result).toBeDefined();
    expect(typeof result.skillsRootPath).toBe("string");
    expect(result.skillsRootPath.length).toBeGreaterThan(0);
  }, 30000);
});

File: golang/tests/pkg/integration/beta_skills_get_metadata_integration_test.go
// ci-stable
// Integration tests for BetaSkills.GetMetadata() via GetSkillMetaData POP action.
//
// Note: This test calls the real backend. Do not run concurrently.
package integration_test

import (
	"testing"

	"github.com/aliyun/wuying-agentbay-sdk/golang/pkg/agentbay"
	"github.com/aliyun/wuying-agentbay-sdk/golang/tests/pkg/agentbay/testutil"
)

func TestBetaSkillsGetMetadata_ReturnsSkillsRootPathAndSkills(t *testing.T) {
	apiKey := testutil.GetTestAPIKey(t)
	agentBay, err := agentbay.NewAgentBay(apiKey)
	if err != nil {
		t.Fatalf("Error initializing AgentBay client: %v", err)
	}

	result, err := agentBay.BetaSkills.GetMetadata()
	if err != nil {
		t.Fatalf("GetMetadata failed: %v", err)
	}

	if result == nil {
		t.Fatal("Expected non-nil result")
	}
	if len(result.SkillsRootPath) == 0 {
		t.Error("Expected non-empty skillsRootPath")
	}
	if result.Skills == nil {
		t.Error("Expected non-nil skills slice")
	}
}

func TestBetaSkillsGetMetadata_WithSkillNamesFilter(t *testing.T) {
	apiKey := testutil.GetTestAPIKey(t)
	agentBay, err := agentbay.NewAgentBay(apiKey)
	if err != nil {
		t.Fatalf("Error initializing AgentBay client: %v", err)
	}

	result, err := agentBay.BetaSkills.GetMetadata(agentbay.GetMetadataOptions{
		SkillNames: []string{"5kvAvffm"},
	})
	if err != nil {
		t.Fatalf("GetMetadata with skillNames failed: %v", err)
	}

	if result == nil {
		t.Fatal("Expected non-nil result")
	}
	if result.Skills == nil {
		t.Error("Expected non-nil skills slice")
	}
	if len(result.SkillsRootPath) == 0 {
		t.Error("Expected non-empty skillsRootPath")
	}
}

func TestBetaSkillsGetMetadata_WithImageId(t *testing.T) {
	apiKey := testutil.GetTestAPIKey(t)
	agentBay, err := agentbay.NewAgentBay(apiKey)
	if err != nil {
		t.Fatalf("Error initializing AgentBay client: %v", err)
	}

	result, err := agentBay.BetaSkills.GetMetadata(agentbay.GetMetadataOptions{
		ImageID: "linux_latest",
	})
	if err != nil {
		t.Fatalf("GetMetadata with imageId failed: %v", err)
	}

	if result == nil {
		t.Fatal("Expected non-nil result")
	}
	if len(result.SkillsRootPath) == 0 {
		t.Error("Expected non-empty skillsRootPath")
	}
}
